src/main/java/org/opensearch/security/configuration/DlsFlsFilterLeafReader.java

```diff
@@ -130,7 +131,8 @@ class DlsFlsFilterLeafReader extends SequentialStoredFieldsLeafReader {
         this.clusterService = clusterService;
         this.auditlog = auditlog;
         this.salt = salt;
-        this.maskedFieldsMap = MaskedFieldsMap.extractMaskedFields(maskFields, maskedFields, salt);
+        this.maskingAlgorithmDefault = clusterService.getSettings().get(ConfigConstants.SECURITY_MASKED_FIELDS_ALGORITHM_DEFAULT);
```

Comment:
QQ, what happens if this setting is not explicitly set? Would it default to blake2b? If so should it not default to blake2b in case any non-supported names are passed in the setting, instead of boot-up failure?

Reply:
For backwards compatibility if not explicitly set it defaults to Blake2b as it currently does. 

In fact I had to raise https://github.com/opensearch-project/security/issues/4274 as I have been asked for it to be backwards compatible with the incorrect parameters that are currently in used in the code in order for the same hashes to be generated.

Reply:
> okay. Is the future expectation that node boot-up should fail if no custom algo is passed?

Not from my point of view. This setting is optional and allows more control over the default algorithm used. It will only prevent node boot if that algorithm is not supported by the JVM.

Reply:
Should we prevent node boot-up and instead display a message stating masking defaulted to blake2b since the custom supplied algo is not recognized by JVM. I'm 50-50 on stopping node boot-up. On one hand, prevent node from coming up would allow users to detect the misconfigured value early but at the same time we are blocking node boot-up for something that can be defaulted to a value, similar to some other settings.

Reply:
Hi @DarshitChanpura Yes, there are arguments either way. I leaned heavily towards stopping boot up due mainly to the following reasons:

- This will be documented under a section that is labelled as Advanced https://opensearch.org/docs/latest/security/access-control/field-masking#advanced-use-an-alternative-hash-algorithm and we could assume the user changing will be an advanced user
- The error message is very clear about why the node doesn't start and should be especially so for an advanced user
- I was concerned that logging a warning and continuing may mean this incompatibility being missed. This may be of particular concern when set for compliance purposes.